docs/lessons/buttons.rst

```diff
@@ -0,0 +1,188 @@
+Buttons
+-------
+
+So far we have created code that makes the device do something. This is called
+*output*. However, we also need the device to react to things. Such things are
+called *input*.
+
+It's easy to remember: output is what the device puts out to the world
+whereas input is what goes into the device for it to process.
+
+The most obvious means of input on the micro:bit are its two buttons, labelled
+``A`` and ``B``. Somehow, we need MicroPython to react to button presses.
+
+This is remarkably simple::
+
+    from microbit import *
+
+    sleep(10000)
+    display.scroll(str(button_a.get_presses()))
+
+All this script does is sleep for ten thousand milliseconds (i.e. 10 seconds)
+and then scrolls the number of times you pressed button ``A``. That's it!
+
+While it's a pretty useless script, it introduces a couple of interesting new
+ideas:
+
+#. The ``sleep`` *function* will make the micro:bit sleep for a certain number
+   of milliseconds. If you want a pause in your program, this is how to do it.
+   A *function* is just like a *method*, but it isn't attached by a dot to an
+   *object*.
+#. There is an object called ``button_a`` and it allows you to get the number
+   of times it has been pressed with the ``get_presses`` *method*.
+
+Since ``get_presses`` gives a numeric value and ``display.scroll`` only
+displays characters, we need to convert the numeric value into a string of
+characters. We do this with the ``str`` function (short for "string" ~ it
+converts things into strings of characters).
+
+The third line is a bit like an onion. If the parenthesis are the
+onion skins then you'll notice that ``display.scroll`` contains ``str`` that
+itself contains ``button_a.get_presses``. Python attempts to work out the
+inner-most answer first before starting on the next layer out. This is called
+*nesting* - the coding equivalent of a Russian Matrioshka doll.
+
+.. image:: matrioshka.jpg
+
+Let's pretend you've pressed the button 10 times. Here's how Python works out
+what's happening on the third line:
```

Comment:
Not in this instance, because there is no code example that immediately follows.